spec/consumer_spec.rb

```diff
@@ -224,6 +224,36 @@
       end
     end
 
+    context "when fetch_batches returns FetchError" do
+      let(:thread_timeout) { 3 }
+      before do
+        allow(cluster).to receive(:mark_as_stale!)
+        allow(consumer).to receive(:fetch_batches).and_raise(Kafka::FetchError)
+      end
+
+      it "runs forever if exit_if is not passed" do
+        th = Thread.new do
+          consumer.each_batch do |message|
+            fail "this block is never called"
+          end
+        end
+
+        expect(th.join(thread_timeout)).to be_nil # join returning nil is a sign of thread not completing within timeout
+        th.kill
+      end
+
+      it "aborts earlier if exit_if is passed" do
+        th = Thread.new do
```

Comment:
not super clear to me why this test needs a new thread? why can't this run inline?

Reply:
These tests are basically copied from https://github.com/Shopify/ruby-kafka/blob/1673aed9078ed23417cf7811ed4b33d1e4e6cac7/spec/consumer_spec.rb#L200-L225 (same file, for a different case when exit_if was introduced)

The thread is required because if the source code changes and `exit_if` was removed, this test will never terminate and loop endlessly. I think @kirs added it here to avoid that.